framework/xds_k8s_testcase.py

```diff
@@ -386,6 +388,40 @@ def removeServerBackends(self, *, server_runner=None):
         # Remove backends from the Backend Service
         self.td.backend_service_remove_neg_backends(neg_name, neg_zones)
 
+    def assertHealthyEndpointsCount(
+        self,
+        test_client,
+        expected_len,
+        msg=None,
+        *,
+        retry_timeout: dt.timedelta = dt.timedelta(minutes=3),
+        retry_wait: dt.timedelta = dt.timedelta(seconds=10),
+        log_level: int = logging.DEBUG,
+    ) -> None:
+        retryer = retryers.constant_retryer(
+            wait_fixed=retry_wait,
+            timeout=retry_timeout,
+            error_note=(
+                f"Timeout waiting for test client {test_client.hostname} to"
+                f" report {expected_len} endpoint(s)."
+            ),
+        )
+        for attempt in retryer:
+            with attempt:
+                client_config = test_client.csds.fetch_client_status_parsed(
+                    log_level=log_level
+                )
```

Comment:
There's a shortcut that abstracts out the source of the data:

```suggestion
                client_config = test_client.get_csds_parsed(log_level=log_level)
```